subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/resolver/DefaultSourcesJarFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.internal.resolver;
+
+import org.gradle.cache.internal.GeneratedGradleJarCache;
+import org.gradle.internal.installation.CurrentGradleInstallation;
+import org.gradle.internal.installation.GradleInstallation;
+
+import java.io.File;
+
+public class DefaultSourcesJarFactory implements SourcesJarFactory {
+
+    private final SourcesJarCreator sourcesJarCreator;
+    private final GeneratedGradleJarCache cache;
```

Comment:
I am not sure if I should be injecting and reusing the generated jar cache here.